src/typrepr.ml

```diff
@@ -167,8 +178,34 @@ module Rec = struct
     ; gen= Some (exp_id ("gen_" ^ typ_name))
     ; spec= Some (exp_id ("spec_" ^ typ_name))
     ; card= Infinite
+    ; boltz_spec= Some (Arbogen.Grammar.Ref typ_name)
     ; of_arbogen= Some (exp_id ("of_arbogen_" ^ typ_name)) }
 
+  let make_generator typs =
+    let loc = Location.none in
+    let grammar =
+      typs
+      |> List.map (fun (name, typ) -> (name, Option.get typ.boltz_spec))
+      |> Arbogen.Frontend.ParseTree.completion
+      |> Arbogen.Frontend.ParseTree.to_grammar
+    in
+    let wg =
+      Arbogen.Boltzmann.(
+        WeightedGrammar.of_grammar (Oracle.Naive.make grammar) grammar)
```

Comment:
You should use `make_expectation :: int -> Grammar.t -> Oracle.t` instead of `make`.
The `int` parameter is the targeted size.

Reply:
:+1: I have not updated this branch after the recent modifications in Arbogen, and I'm planning to change some other minor stuff too. I hoping to finish today so that we can merge this soon